signals_unix.go

```diff
@@ -19,25 +19,16 @@
 package containerd
 
 import (
-	"fmt"
-	"strconv"
-	"strings"
 	"syscall"
 
-	"golang.org/x/sys/unix"
+	"github.com/moby/sys/signal"
 )
 
 // ParseSignal parses a given string into a syscall.Signal
 // the rawSignal can be a string with "SIG" prefix,
 // or a signal number in string format.
 func ParseSignal(rawSignal string) (syscall.Signal, error) {
-	s, err := strconv.Atoi(rawSignal)
-	if err == nil {
-		return syscall.Signal(s), nil
-	}
-	signal := unix.SignalNum(strings.ToUpper(rawSignal))
-	if signal == 0 {
-		return -1, fmt.Errorf("unknown signal %q", rawSignal)
-	}
-	return signal, nil
+	// Go's sys/unix doesn't support real-time signals (e.g. SIGRTMIN+3).
+	// Use moby/sys/signal instead.
```

Comment:
I see the Windows variant in the containerd repo has some signals that were not yet in the moby/sys version (.https://github.com/containerd/containerd/blob/main/signals_windows.go#L28) Perhaps would be good to add the changes there, then after that we could just make this an alias (potentially deprecate), and there would be no need for separate windows/unix variants in this repo

Reply:
Yeah. I am uncertain about Windows and didn't touch that in this PR.

Getting rid of signals_windows.go sounds great, but there is #4793 would change the scope of signals_*.go and possibly moby/sys/signals. (cc @katiewasnothere)

Reply:
At least for the signals itself, I think the goal should be to have a canonical list. I haven't looked at the other utilities in those files (on mobile currently), and where the most logical place for those would be.

Reply:
Sounds good to me. Adding them in moby/sys. https://github.com/moby/sys/pull/71